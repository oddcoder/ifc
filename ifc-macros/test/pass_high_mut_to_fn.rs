use ifc_macros::ifc_block;

fn inc(x: &mut usize) {
    *x += 1;
}
fn main() {
    ifc_block!{
        let x : usize = 5;
        inc(x);
    };
}