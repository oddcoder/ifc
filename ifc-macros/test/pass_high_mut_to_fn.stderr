error: Cannot pass high expression as an argument to functions
 --> $DIR/pass_high_mut_to_fn.rs:9:9
  |
9 |         inc(x);
  |         ^^^^^^
  |
help: High Expression
 --> $DIR/pass_high_mut_to_fn.rs:9:13
  |
9 |         inc(x);
  |             ^
